src/remote/remote.ts

```diff
@@ -398,256 +393,262 @@ export class Remote {
-		// Coder binary properly.
-		//
-		// If we didn't write to the SSH config file, connecting would fail with
-		// "Host not found".
-		try {
-			this.logger.info("Updating SSH config...");
-			await this.updateSSHConfig(
-				workspaceClient,
-				parts.label,
-				parts.host,
-				binaryPath,
-				logDir,
-				featureSet,
-			);
-		} catch (error) {
-			this.logger.warn("Failed to configure SSH", error);
-			throw error;
-		}
+			const logDir = this.getLogDir(featureSet);
 
-		// TODO: This needs to be reworked; it fails to pick up reconnects.
-		this.findSSHProcessID().then(async (pid) => {
-			if (!pid) {
-				// TODO: Show an error here!
-				return;
-			}
-			disposables.push(this.showNetworkUpdates(pid));
-			if (logDir) {
-				const logFiles = await fs.readdir(logDir);
-				const logFileName = logFiles
-					.reverse()
-					.find(
-						(file) => file === `${pid}.log` || file.endsWith(`-${pid}.log`),
-					);
-				this.commands.workspaceLogPath = logFileName
-					? path.join(logDir, logFileName)
-					: undefined;
-			} else {
-				this.commands.workspaceLogPath = undefined;
+			// This ensures the Remote SSH extension resolves the host to execute the
+			// Coder binary properly.
+			//
+			// If we didn't write to the SSH config file, connecting would fail with
+			// "Host not found".
+			try {
+				this.logger.info("Updating SSH config...");
+				await this.updateSSHConfig(
+					workspaceClient,
+					parts.label,
+					parts.host,
+					binaryPath,
+					logDir,
+					featureSet,
+				);
+			} catch (error) {
+				this.logger.warn("Failed to configure SSH", error);
+				throw error;
 			}
-		});
 
-		// Register the label formatter again because SSH overrides it!
-		disposables.push(
-			vscode.extensions.onDidChange(() => {
-				disposables.push(
-					this.registerLabelFormatter(
+			// TODO: This needs to be reworked; it fails to pick up reconnects.
+			this.findSSHProcessID().then(async (pid) => {
+				if (!pid) {
+					// TODO: Show an error here!
+					return;
+				}
+				disposables.push(this.showNetworkUpdates(pid));
+				if (logDir) {
+					const logFiles = await fs.readdir(logDir);
+					const logFileName = logFiles
+						.reverse()
+						.find(
+							(file) => file === `${pid}.log` || file.endsWith(`-${pid}.log`),
+						);
+					this.commands.workspaceLogPath = logFileName
+						? path.join(logDir, logFileName)
+						: undefined;
+				} else {
+					this.commands.workspaceLogPath = undefined;
+				}
+			});
+
+			// Register the label formatter again because SSH overrides it!
+			disposables.push(
+				vscode.extensions.onDidChange(() => {
+					// Dispose previous label formatter
+					labelFormatterDisposable.dispose();
+					labelFormatterDisposable = this.registerLabelFormatter(
 						remoteAuthority,
 						workspace.owner_name,
 						workspace.name,
 						agent.name,
-					),
-				);
-			}),
-		);
+					);
+					disposables.push(labelFormatterDisposable);
```

Comment:
Probably not a big deal, but in theory we could collect an unbounded number of disposed label formatters in the disposables array.  Maybe we should store a single function that disposes the current formatter?  Something like:

```javascript
disposables.push({
  dispose: () => labelFormatterDisposable.dispose()
})
```